src/librustc_typeck/diagnostics.rs

```diff
@@ -4567,6 +4567,52 @@ i_am_a_function();
 ```
 "##,
 
+E0619: r##"
+A not (yet) known type was used.
+
+Erroneous code example:
+
+```compile_fail,E0619
+let x;
+
+match x {
+    (..) => {} // error: the type of this value must be known in this context
+    _ => {}
+}
+```
+
+To fix this error, just specify the type of the variable. Example:
+
+```
+let x: i32 = 0; // Here, we say that `x` is an `i32` (and give it a value to
+                // avoid another compiler error).
+
+match x {
+    0 => {} // ok!
+    _ => {}
+}
+```
+"##,
+
+E0620: r##"
+A cast to an unsized type was attempted.
+
+Erroneous code example:
+
+```compile_fail,E0620
+let x = &[1_usize, 2] as [usize]; // error: cast to unsized type: `&[usize; 2]`
+                                  //        as `[usize]`
+```
+
+In Rust, some types don't have a size at compile-time (like slices and traits
```

Comment:
Maybe something like this?

In Rust, some types don't have a don't have a known size at compile-time.
For example, in a slice type like `[u32]`, the number of elements is not known
at compile-time and hence the overall size cannot be computed. As a result,
such types can only be manipulated through a reference (e.g., `&T` or `&mut T`)
or other pointer-type (e.g., `Box` or `Rc`). Try casting to a reference instead:

```
let x = &[1_usize, 2] as &[usize]; // ok!
```